### ISSUE TITLE
documentation

### ISSUE THREAD

**Issue by davidedc** [davidedc/livecodelab#260]
what people are looking for:
![screen shot 2015-06-23 at 10 10 09](https://cloud.githubusercontent.com/assets/769568/8302694/572c2800-1990-11e5-8d97-a2e463681fb4.png)

what they get:
![screen shot 2015-06-23 at 10 14 30](https://cloud.githubusercontent.com/assets/769568/8302741/c2cf9cd6-1990-11e5-96b8-61a84b43bd07.png)

**Comment by rumblesan** [comment]
haha, so yeah, we could do with improving this couldn't we. Maybe worth making use of the wiki that github provides for a project?

**Comment by ashleyjamesbrown** [comment]
Im teaching this with kids aged 9/10/11 and blatantly a reference would be awesome.

**Comment by davidedc** [comment]
+1 . Argh yes just didn't have the time to make it happen...

**Comment by ashleyjamesbrown** [comment]
Are all the methods easily understandable from the source code ? Ive been using processing for 8 years and also p5js for about a year so im fairly up on the potential syntax and usage. Happy to try and find time for some basic beginner reference. I just know that it would help a lot.

**Comment by davidedc** [comment]
hi, that would be ace.
I don't think there is a need to dive into the source code - the commands are really either same as Processing or they are illustrated in the tutorial. All the keywords are in the first 90 lines here: https://github.com/davidedc/livecodelab/blob/master/src/coffee/languages/livelangv1/code-preprocessor.coffee#L70

Also note that we have this: https://github.com/davidedc/livecodelab/blob/009b5a9b92fe5e55b4d123c8b840c56f372a7a25/docs/intros/functional-aspects.md and this: https://github.com/davidedc/livecodelab/blob/009b5a9b92fe5e55b4d123c8b840c56f372a7a25/docs/intros/language-intro.md

and of course all the tutorial menu could be of help...

**Comment by davidedc** [comment]
Couple more notes.

1- Worth checking some "guerrilla notebook" solutions such as:

- kajero http://www.joelotter.com/kajero/ . Seems like really quick and no-fuss, I'd love this one. See some notes on it here: https://www.reddit.com/r/programming/comments/4jhaav/kajero_interactive_javascript_notebooks/
- tiddlywiki http://tiddlywiki.com/

as these would allow very simple hosting (it's all static content) and also give editing ability without needed complex server solutions (just edit, then export and save the export into github).

Also there are some discussions around document generation here: http://stackoverflow.com/questions/35623309/tool-used-to-generate-three-js-documentation (three.js docs are nice because they have an "edit" button that points directly to the source).

2- the place to add documentation would ideally be a section on the website ( livecodelab.net ) , so anybody can access it easily . The source code for the site is kept separate, here: https://github.com/davidedc/Livecodelab-website/tree/gh-pages

3- we should ideally add a version in the URL e.g. /reference/2.1/ (and /reference/latest/ pointing at the latest version) since we are definitely going to break things with time...

**Comment by Luraktinus** [comment]
how about a tutorial for making music and a reference for the samples that i could use

**Comment by davidedc** [comment]
@Luraktinus a couple of the examples really do contain the whole syntax and samples, let me know if those alone suffice!

**Comment by Luraktinus** [comment]
yeah, i have seen this but i kinda miss a list of samples that are included (not some of it, all of them)

**Comment by davidedc** [comment]
the “list of all samples” really should contain all of them. If it doesn’t, it’s a mistake, perhaps you could tell us the ones that are missing ?

**Comment by Luraktinus** [comment]
to make sure we are on the same list, please tell me what list are you talking about...

**Comment by Luraktinus** [comment]
if you look at the samples in here, there are some more.
https://github.com/davidedc/livecodelab/blob/master/src/programs/demos/sounddjcastro.lcl.yaml

**Comment by davidedc** [comment]
yes we should add those in a next version. Feel free to submit a patch if you get to it before anyone else!
